Automattic/Automattic-Tracks-Android
# [Build] Upgrade Gradle to 8.2.1 and AGP to 8.1.0

Platform Request: `pdnsEh-13V-p2`
Project Thread: `paaHJt-57Z-p2`

This PR upgraded Gradle to [8.2.1](https://docs.gradle.org/8.2.1/release-notes.html) and AGP to [8.0.2](https://developer.android.com/build/releases/gradle-plugin#agp-8-0-2).

PS: This Gradle and AGP upgrade was done in order to align this repo with the rest of such [WordPress-Mobile](https://github.com/wordpress-mobile) Android library related repos. However, mind the fact that this work here is not necessary to complete the aforementioned project, just some extra work to help with the wider alignment efforts.

-----

🚫 Blocked By:


<details>
    <summary>CI Failure</summary>

```
[2023-07-18T13:04:04Z] FAILURE: Build failed with an exception.
--
  | [2023-07-18T13:04:04Z]
  | [2023-07-18T13:04:04Z] * Where:
  | [2023-07-18T13:04:04Z] Build file '/var/lib/buildkite-agent/builds/ci-android-i-098082138775f5d9d-1/automattic/automattic-tracks-android/AutomatticTracks/build.gradle' line: 2
  | [2023-07-18T13:04:04Z]
  | [2023-07-18T13:04:04Z] * What went wrong:
  | [2023-07-18T13:04:04Z] An exception occurred applying plugin request [id: 'com.android.library', version: '8.0.2']
  | [2023-07-18T13:04:04Z] > Failed to apply plugin 'com.android.internal.library'.
  | [2023-07-18T13:04:04Z]    > Android Gradle plugin requires Java 17 to run. You are currently using Java 11.
  | [2023-07-18T13:04:04Z]       Your current JDK is located in /usr/lib/jvm/java-11-openjdk-11.0.19.0.7-1.amzn2.0.1.x86_64
  | [2023-07-18T13:04:04Z]       You can try some of the following options:
  | [2023-07-18T13:04:04Z]        - changing the IDE settings.
  | [2023-07-18T13:04:04Z]        - changing the JAVA_HOME environment variable.
  | [2023-07-18T13:04:04Z]        - changing `org.gradle.java.home` in `gradle.properties`.
```

</details>

-----

Breaking Change List:

1. [Move package from android manifest to build files](https://github.com/Automattic/Automattic-Tracks-Android/commit/52a321b3aba07a0ddd3a6d4f07fa99b5b214493f)
2. [Enable build config build feature for editor and wordpress](https://github.com/Automattic/Automattic-Tracks-Android/commit/89e0f8036999d7ecf6e18e8f16b9d0fa944e9613)
3. [Fix compile java and kotlin jvm target compat. build failure](https://github.com/Automattic/Automattic-Tracks-Android/commit/e429fe262e4a9d035c9d28c30e348a44e34140d3)

AGP Upgrade Assistant List:

1. [Remove ignored configuration properties](https://github.com/Automattic/Automattic-Tracks-Android/commit/b03997b1e854c703a175b9d43b34a4524f4d8b4e)
2. [Migrate dsl properties from lint options to lint](https://github.com/Automattic/Automattic-Tracks-Android/commit/57d1c60b045a2ce00de06d64396763ca91ae130f)

Gradle Configuration List:

1. [Update org gradle jvmargs gradle property to xmx 6g](https://github.com/Automattic/Automattic-Tracks-Android/commit/40323d83c3147d910daef7e4f3cd20993b3d870b)

Lint Configuration List:

1. [Configure lint on all modules to adhere to lint xml file](https://github.com/Automattic/Automattic-Tracks-Android/commit/4a002366835c6be70f47bfb7003c2656f4faf110)
2. [Move lint and baseline xml files to dedicated config lint dir](https://github.com/Automattic/Automattic-Tracks-Android/commit/551a906f730129630974d857fd4567615fe76ef5)

Dependency Versions Refactor List:

1. [Extract all sdk version to a separate ext block](https://github.com/Automattic/Automattic-Tracks-Android/commit/2aab0ae54e3cb8b9c78d76d199f91327bd7a8ef2)
2. [Extract sentry bom version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/c7a3118f10b96413235209221edb43c05c58ed91)
3. [Extract androidx annotation version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/33e8083db7a9baa8acc2e60cd68b13bfae9aa355)
4. [Extract squareup okhttp version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/6f970fd6630ca93dd1c9475004e4aed97e7cfe00)
5. [Extract kotlinx coroutines version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/93c349f79dbaabd978270aa7eab091a37283f899)
6. [Extract junit version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/06ff595cafe32e9259b75011ebcf4224f2fd933c)
7. [Extract assertj version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/3a08840a69dd9c622aa287496c2a3c69e9ba2049)
8. [Extract mockito kotlin version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/f77277b566ebecaa3352d21d1d43288611ae4156)
9. [Extract wordPress fluxc version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/9e7390e7a759a89ed6e6b2fbda8998cd780bc8b1)
10. [Extract wordPress utils version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/bdb2ba4d6a94ca9a509535735f8c5850fc96fef3)
11. [Extract androidx core version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/665a807c7556a2e0eca15527ed4404a12869e6a1)
12. [Extract androidx appcompat version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/58a5738640526cc0063cf572189a54cc27de7453)
13. [Extract google material version to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/f0ba5b409f3baf3ea63a581a174496d39a5d7641)
14. [Add versions groups to root build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/3aed28b1951d23b4020396f5f253d4f87dd68f4e)
15. [Add com android application plugin to plugin management](https://github.com/Automattic/Automattic-Tracks-Android/commit/249ae7421f19f7c1bc18ab247bbb7b0259fe5407)
16. [Extract agp version to settings build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/0f8a774ef994e9da40d1952c9ffebcb610c9589c)
17. [Extract kotlin version to settings build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/3b7278553abc7727ce2e73b7bc2aec9f8d9e6df4)
18. [Extract automattic publish to s3 version to settings build gradle](https://github.com/Automattic/Automattic-Tracks-Android/commit/7efbf8836490e2163ddea6928da3db6f2f7c7d5d)

-----

## Associated clients (`N/A`)

FYI: `Automattic-Tracks-Android` is not part of a composite build, neither for `WPAndroid`, `WCAndroid` clients or the `FluxC` library.

-----

## To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could run `./gradlew build` locally and verify that  it is successful (see [previous](https://github.com/Automattic/Automattic-Tracks-Android/pull/135) such update).

-----

## Merge instructions

- [ ] In coordination with all the other library and client PRs, merge this PR only when every PR is ready to be merged. FYI: This is a nice to have step here, but not a necessary step per se since `wellsql` is not part of a composite build.
